implementations/python/sugar-lift-python-source/src/sugar_lift_python_source/dependency_export_adapter.py

```
@@ -129,12 +129,50 @@ def _resolve_export_uncached(
     tree = parsed_tree(module.source, module.source_seat)
-    binding = _export_block(tree.body, exported_name, None)
+    binding, locus = _export_block_with_locus(tree.body, exported_name, None)
     dynamic_getattr = any(
         isinstance(node, (ast.FunctionDef, ast.AsyncFunctionDef))
         and node.name == "__getattr__"
         for node in tree.body
     )
+    # Normal-completion authority for the prefix (statements strictly before
+    # the unique export-binding locus): producer-owned Completed faces only.
+    if locus is not None and not _prefix_has_completed_fallthrough(module, locus):
+        return _gap(
```

Ruff (0.16.0)

[error] Undefined name `_gap`

(F821)